### PR TITLE
New feature: Option to use issue import preview API

### PR DESCRIPTION
**Pull request by mdbenito**
This PR:
* adds an option to use the preview API for issue imports. This enables setting the creation and modification date of issues and comments.
* documents it
* adds `settings.github.token_owner`. Required to check ownership of comments against the user creating the new content on GH.

Additionally, it:

* caches a call to getAllBranches()
* Fixes a typing problem in https://github.com/piceaTech/node-gitlab-2-github/blob/7572017dfbcb0a7a2a8917ab6cb99ebe2c173877/src/githubHelper.ts#L858
* Fixes a problem whereby labels are not converted to lowercase before using them in `createIssue()`: https://github.com/piceaTech/node-gitlab-2-github/blob/7572017dfbcb0a7a2a8917ab6cb99ebe2c173877/src/githubHelper.ts#L258

### CODE REVIEW

**Comment by mdbenito** [comment]
Sorry for merging a couple of fixes into the PR, but I thought them to be minor enough not to warrant separate PRs.

File: src/githubHelper.ts [hunk 1]
```diff
@@ -280,6 +298,180 @@ export default class GithubHelper {
...
+    //
+    // Issue Attachments
+    //
+
+    // if the issue contains a url that contains "/uploads/", it is likely to
+    // have an attachment. Therefore, add the "has attachment" label.
+    if (props.body && props.body.indexOf('/uploads/') > -1 && !settings.s3) {
+      props.labels.push('has attachment');
+    }
+    await utils.sleep(this.delayInMs);
```
Comment (spruce) [inline src/githubHelper.ts:378]: Is it neccessary to sleep here?

File: src/githubHelper.ts [hunk 2]
```diff
@@ -586,6 +785,11 @@ export default class GithubHelper {
     }
 
     // Failing all else, create an issue with a descriptive title
+
+    let userIsCreator = pullRequest.author && 
+      ((settings.usermap && settings.usermap[pullRequest.author.username] === settings.github.token_owner) ||
+       (pullRequest.author.username === settings.github.token_owner));
```
Comment (spruce) [inline src/githubHelper.ts:791]: You check that multiple times. Doesn't it make sense to extract it into its own function?

**Review by spruce — COMMENTED** [review]
Looks nice. Thanks for work :)

File: src/githubHelper.ts (hunk 2 above)
Comment (mdbenito) [reply src/githubHelper.ts:791]: yes, it is checked twice and I think I even factored it out into a function but it wasn't exactly the same functionality, then I simplified it and now it is indeed the same 😅 I'll fix that too.

**Comment by mdbenito** [comment]
I addressed your points and fixed the label case conversion to only happen if `settings.conversion.useLowerCaseLabels` is true. I am unsure about the return when `settings.debug` is true, though, but it is exactly the same behaviour as in the previous `createIssue`, so that should be ok.

**Pull request merged**